powerline/segments/vim/plugin/ale.py

```diff
@@ -0,0 +1,97 @@
+# vim:fileencoding=utf-8:noet
+"""
+Asynchronous Lint Engine (ALE) plugin
+
+Largely adapted from Sytastic plugin
+"""
+
+from __future__ import (
+	unicode_literals, division, absolute_import, print_function
+)
+try:
+	import vim
+except ImportError:
+	vim = object()
+from powerline.bindings.vim import vim_global_exists
+from powerline.theme import requires_segment_info
+
+
+# To turn on logging, in the Powerline configuration fie config.json,
+# in the "common" key, add log_file and log_level key-value pairs as
+# appropriate and desired, for example:
+#
+# "common": {
+#	  "term_truecolor": false,
+#	  "log_file": "/mylogdir/powerline.log",
+#	  "log_level": "INFO"
+# },
+#
+# and then turn on debug internal variable to True
+@requires_segment_info
+def ale(
+	segment_info,
+	pl,
+	err_format='ERR:  {first_line} ({num}) ',
+	warn_format='WARN:  {first_line} ({num}) '
+):
+	"""
+	Show whether Asynchronous Lint Engine (ALE) has found any errors
+	or warnings
+
+	:param str err_format:
+		Format string for errors.
+
+	:param str warn_format:
+		Format string for warnings.
+
+	Highlight groups used: ``ale:warning`` or
+	``warning``, ``ale:error`` or ``error``.
+	"""
+	# pylint: disable=C0103,W0613
+	debug = False
+	bufnr = str(segment_info['bufnr'])
+	ale_enabled_exists = vim_global_exists('ale_enabled')
+	ale_enabled = ale_enabled_exists and int(vim.eval('g:ale_enabled'))
+	if debug:
```

Comment:
This should not be hidden under `if debug`, you should use `pl.debug` instead.

Reply:
Actually removed all debug logging, in hindsight this was only needed to get the ALE interfacing correct, and it only clutters code now. Removal also makes Syntastic and ALE plugin files more similar.